app/src/main/java/com/kms/cura/view/activity/MessageThreadActivity.java

```diff
@@ -200,4 +219,105 @@ public ListView getLvMessage() {
     public boolean isMultiMode() {
         return multiMode;
     }
+
+    private void deleteMessages(final ArrayList<MessageEntity> entities) {
+        new AsyncTask<Object, Void, Void>() {
+            private Exception exception = null;
+            ProgressDialog pDialog;
+
+            @Override
+            protected void onPreExecute() {
+                pDialog = new ProgressDialog(MessageThreadActivity.this);
+                pDialog.setMessage(getString(R.string.loading));
+                pDialog.setCancelable(false);
+                pDialog.show();
+            }
+
+            @Override
+            protected Void doInBackground(Object[] params) {
+                try {
+                    for (MessageEntity entity : entities) {
+                        MessageController.deleteMessage(CurrentUserProfile.getInstance().getEntity(), entity);
+                    }
+                    MessageController.loadMessage(CurrentUserProfile.getInstance().getEntity());
+                    messageEntities = MessageController.getMessageByThread(CurrentUserProfile.getInstance().getEntity(), getIntent().getStringExtra(MessageThreadEntity.CONVERSATION));
+                } catch (Exception e) {
+                    exception = e;
+                }
+                return null;
+            }
+
+            @Override
+            protected void onPostExecute(Void aVoid) {
+                pDialog.dismiss();
+                if (exception != null) {
+                    ErrorController.showDialog(MessageThreadActivity.this, "Error : " + exception.getMessage());
+                } else {
+                    if (messageEntities == null) {
+                        backToMain();
+                    } else {
+                        adapter.setData(messageEntities);
+                        adapter.notifyDataSetChanged();
+                        Toast.makeText(MessageThreadActivity.this, R.string.deleted, Toast.LENGTH_SHORT).show();
+                        sendRefreshRequest();
+                    }
+                }
+            }
+        }.execute();
+    }
+
+    private void refreshMessage() {
+        new AsyncTask<Object, Void, Void>() {
+            private Exception exception = null;
+
+            @Override
+            protected Void doInBackground(Object[] params) {
+                try {
+                    MessageController.loadMessage(CurrentUserProfile.getInstance().getEntity());
```

Comment:
Same here, extract CurrentUserProfile.getInstance().getEntity() to variable for reuse